File: ki-aim-frontend/src/app/shared/model/data-type.ts
export enum DataType {
    BOOLEAN,
	DATE_TIME,
	DECIMAL,
	INTEGER,
	STRING,
	DATE,
	UNDEFINED
}

export function dataTypeFromString(value: String): DataType {
	switch (value) {
		case "BOOLEAN": return DataType.BOOLEAN
		case "DATE_TIME": return DataType.DATE_TIME
		case "DECIMAL": return DataType.DECIMAL
		case "INTEGER": return DataType.INTEGER
		case "STRING": return DataType.STRING
		case "DATE": return DataType.DATE
		default: return DataType.UNDEFINED
	}
}

File: src/main/java/de/kiaim/platform/model/data/DataType.java
package de.kiaim.platform.model.data;

import de.kiaim.platform.model.data.configuration.DataScale;
import lombok.Getter;

@Getter
public enum DataType {
	BOOLEAN(DataScale.NOMINAL),
	DATE_TIME(DataScale.INTERVAL),
	DECIMAL(DataScale.RATIO),
	INTEGER(DataScale.INTERVAL),
	STRING(DataScale.NOMINAL),
	DATE(DataScale.INTERVAL),
	UNDEFINED(DataScale.NOMINAL);

	DataType(final DataScale defaultScale) {
		this.defaultScale = defaultScale;
	}

	private final DataScale defaultScale;
}
